sdk/core/Azure.Core/src/Delay.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Strategy to control delay behavior.
+    /// </summary>
+#pragma warning disable AZC0012 // Avoid single word type names
+    public abstract class Delay
+#pragma warning restore AZC0012 // Avoid single word type names
+    {
+        private readonly Random _random = new ThreadSafeRandom();
+        private readonly double _minJitterFactor;
+        private readonly double _maxJitterFactor;
+        private readonly TimeSpan _maxDelay;
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="maxDelay"></param>
+        /// <param name="jitterFactor"></param>
+        protected Delay(TimeSpan? maxDelay = default, double jitterFactor = 0.2)
+        {
+            // use same defaults as RetryOptions
+            _minJitterFactor = 1 - jitterFactor;
+            _maxJitterFactor = 1 + jitterFactor;
+            _maxDelay = maxDelay ?? TimeSpan.FromMinutes(1);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="initialDelay"></param>
+        /// <param name="maxDelay"></param>
+        /// <returns></returns>
+        public static Delay CreateExponentialDelay(
+            TimeSpan? initialDelay = default,
+            TimeSpan? maxDelay = default)
+        {
+            return new ExponentialDelay(initialDelay ?? TimeSpan.FromSeconds(0.8), maxDelay ?? TimeSpan.FromMinutes(1));
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="delay"></param>
+        /// <returns></returns>
+        public static Delay CreateFixedDelay(
+            TimeSpan? delay = default)
+        {
+            return new FixedDelay(delay ?? TimeSpan.FromSeconds(0.8));
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="response"></param>
+        /// <param name="retryNumber"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        protected abstract TimeSpan GetNextDelayCore(Response? response, int retryNumber, IDictionary<string, object?> context);
```

Comment:
Instead of the IDictionary<string, object?> here I wonder if we can achieve per set and global state in both LRO and Retry by adding 2 ctors to RetryPolicy

`public RetryPolicy(..., Delay delay = null)`
and
`public RetryPolicy(..., Func<Delay> delayFactory)`

If a user wanted to have per set state in a retry they would just pass in a factory

```c#
options.RetryPolicy = new RetryPolicy(..., () => new MyCustomDelay());
```

I think this gives us a universal way to handle state across retry and lro and lets the customer deal with a typed state vs always needing to cast from `object?` out of the dictionary.

Reply:
I would be fine with this approach. The factory approach would also be additive so we could wait to add the new ctor if we wanted.